src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java

```diff
@@ -185,6 +193,67 @@ public AwsCredential getControlledAwsS3StorageFolderCredential(
         "Error getting AWS S3 Storage Folder credential.");
   }
 
+  // TODO(BENCH-598) move these to CRL / Axon
+
+  public static S3Client getS3Client(AwsCredential awsCredential, String region) {
+    return S3Client.builder()
+        .region(Region.of(region))
+        .credentialsProvider(
+            StaticCredentialsProvider.create(
+                AwsSessionCredentials.create(
+                    awsCredential.getAccessKeyId(),
+                    awsCredential.getSecretAccessKey(),
+                    awsCredential.getSessionToken())))
+        .build();
+  }
+
+  public Integer getNumObjects(UUID workspaceId, AwsS3StorageFolder awsStorageFolder, long limit) {
+    S3Client s3Client =
+        getS3Client(
+            getControlledAwsS3StorageFolderCredential(
+                workspaceId,
+                awsStorageFolder.getId(),
+                AwsCredentialAccessScope.READ_ONLY,
+                CREDENTIAL_EXPIRATION_SECONDS_DEFAULT),
+            awsStorageFolder.getRegion());
+
+    try {
+      Iterator<ListObjectsV2Response> listIterator =
+          s3Client
+              .listObjectsV2Paginator(
+                  ListObjectsV2Request.builder()
+                      .bucket(awsStorageFolder.getBucketName())
+                      .prefix(awsStorageFolder.getPrefix() + "/")
+                      .maxKeys(S3_CLIENT_RESULTS_PER_CALL)
```

Comment:
Should `maxKeys` be `Math.min(limit, S3_CLIENT_RESULTS_PER_CALL)`?

Reply:
If I remember correctly the `limit` was put in place to make the delete time checks run faster, so I think that's probably the primary consideration here: https://github.com/DataBiosphere/terra-cli/pull/162

Reply:
makes sense, done